packages/grafana-data/src/transformations/transformers/calculateField.ts

```diff
@@ -98,10 +103,18 @@ export const calculateFieldTransformer: DataTransformerInfo<CalculateFieldTransf
           creator = getBinaryCreator(defaults(binaryOptions, defaultBinaryOptions), data);
         } else if (mode === CalculateFieldMode.Index) {
           return data.map((frame) => {
+            const indexArr = [...Array(frame.length).keys()];
+
+            if (options.index?.showPercentage) {
+              for (let i = 0; i < indexArr.length; i++) {
+                indexArr[i] = (indexArr[i] / indexArr.length) * 100;
```

Comment:
What happens to numbers not cleanly divisible by the total? Like, what does (3/11) * 100 render for a value?

Reply:
It should show the resulted float value as a percentage. Is there any specific case you have in mind?

Reply:
I just didn't see anywhere it was restricting the significant digits after the decimal, so like 3/11 = .2727272727... would be 27.27272727%? Or does it cut it to 2 places somewhere?

Reply:
Good question! I think it gets rounded in the panel but would need to double check. I'll look into it. Thanks @baldm0mma!